packages/flutter/lib/src/material/search.dart

```diff
@@ -199,7 +199,9 @@ abstract class SearchDelegate<T> {
     final ThemeData theme = Theme.of(context)!;
     assert(theme != null);
     return theme.copyWith(
-      primaryColor: Colors.white,
+      primaryColor: theme.brightness == Brightness.dark
```

Comment:
This use of `primaryColor` in dark themes is something that @HansMuller is trying to sort out at the moment, so this may need to change in the future. For now though this is a reasonable fix for this case. At some point this should be migrated to colorScheme colors.

Reply:
Got it, thanks for the heads up. Hans, let me know if I should open an issue for this or comment anywhere so it's not missed.